### ISSUE TITLE
[KettlerUSB] port is not open, retry in 10s

### ISSUE THREAD

**Issue by laurent-web** [360manu/kettlerUSB2BLE#7]
Hello,
I've this message when i start  , after sudo node server.js
it 's the code from file kettlerUSB.JS, generate by line 160  

internalOpen() {
		this.port.open((err) => {
			if (!err)
				return;
			console.log('[KettlerUSB] port is not open, retry in 10s');
			setTimeout(() => this.internalOpen(), 10000);
		});
	};

i don't know how to correct this error,
the access at browser pi-adress:3000  is ok
when i start zwift, it recognize kettlerBLE, but when i run, nothing   0 watts on browser eton zwift.
Is someone can help me?
Thank you.
Laurent

**Comment by 360manu** [comment]
Hello

It seems that the soft is running.
Very strange !

Some ideas ::
Have you been able to connect to the bike using your PC ?
With GoldenCheetah or KettlerWordTour ?

Are you using a Pi Zero ?
The COM port may have a different name on other models.
 
you can also log the error:
add this : 
console.log(err);
console.log('[KettlerUSB] port is not open, retry in 10s');

**Comment by laurent-web** [comment]
![20200617_105447](https://user-images.githubusercontent.com/67018090/84884240-c3350600-b091-11ea-9ce4-779e0516cba5.jpg)
![20200617_105425](https://user-images.githubusercontent.com/67018090/84884270-ccbe6e00-b091-11ea-8c40-947e7868af11.jpg)

Hello,
thank you for your answer.
it 's a rapsberry pi zero WH.
i can connect to KettlerWordTour . it's running good.
i join screenshot. 
i'll try the console.log.
laurent

**Comment by dmitriimi** [comment]
Hello!
I bought an RPi ZERO WH for this.
I get the same problem.

```
pi@raspberrypi:~/kettlerUSB2BLE $ sudo node server.js
ENOENT: no such file or directory, open '/dev/i2c-1'
[BikeState starting]
[KettlerUSB] constructor
[KettlerUSB] open
[KettlerBLE starting]
Kettler app listening on port 3000!
[KettlerUSB] port is not open, retry in 10s
[KettlerBLE stateChange] new state: poweredOn
[KettlerBLE advertisingStart] success
[KettlerBLE servicesSet] success
[KettlerBLE setServices] success
[KettlerUSB] port is not open, retry in 10s
[KettlerUSB] port is not open, retry in 10s
[KettlerUSB] port is not open, retry in 10s
[KettlerUSB] port is not open, retry in 10s

```

**Comment by 360manu** [comment]
Hello

I think the problem is coming for the oled screen driver.
I added a screen to display the slope and gears.
It's not optional at this stage and it was sendind an exception line 100 or
103.
I removed it from the code.

Tell me if it's ok

Emmanuel

**Comment by dmitriimi** [comment]
Same "port is not open, retry in 10s" 

```
pi@raspberrypi:~/kettlerUSB2BLE $ sudo node server.js
[BikeState starting]
[KettlerUSB] constructor
[KettlerUSB] open
[KettlerBLE starting]
Kettler app listening on port 3000!
[KettlerUSB] port is not open, retry in 10s
[KettlerBLE stateChange] new state: poweredOn
[KettlerBLE advertisingStart] success
[KettlerBLE servicesSet] success
[KettlerBLE setServices] success
[KettlerUSB] port is not open, retry in 10s
[KettlerUSB] port is not open, retry in 10s
```

**Comment by laurent-web** [comment]
[
![20200620_103136](https://user-images.githubusercontent.com/67018090/85198806-aac02800-b2eb-11ea-87b2-4f52c2c4a1d7.jpg)
![20200620_110514](https://user-images.githubusercontent.com/67018090/85198810-ae53af00-b2eb-11ea-84fa-7e13662b93e0.jpg)
![20200620_110605](https://user-images.githubusercontent.com/67018090/85198815-b14e9f80-b2eb-11ea-90c6-958b6ce92ea1.jpg)
![20200620_111946](https://user-images.githubusercontent.com/67018090/85198818-b3186300-b2eb-11ea-8ff5-f5f0fc869446.jpg)
![20200620_112058](https://user-images.githubusercontent.com/67018090/85198820-b4e22680-b2eb-11ea-9191-6fe303ab2be7.jpg)

Hello,
i also try (I2C not enabled in the first try)
i connect first with screen usb socket.
laurent





](url)

**Comment by laurent-web** [comment]
[
![20200620_113032](https://user-images.githubusercontent.com/67018090/85198931-58333b80-b2ec-11ea-9a3e-0fc791a2e9e3.jpg)
![20200620_113038](https://user-images.githubusercontent.com/67018090/85198937-5bc6c280-b2ec-11ea-8789-4c60f6953dbc.jpg)
![20200620_113152](https://user-images.githubusercontent.com/67018090/85198938-5d908600-b2ec-11ea-85fd-ec04e141dfbe.jpg)

i enabled I2C for the 2nd try and connect the usb cable at the back of the screen to the raspberry.
laurent

**Comment by laurent-web** [comment]
Hello,

Before starting node, i try to configure with minicom the ttyUSB0 port.
i only change bauds (9600). i don't change the other settings.
I start node and i hear a sound from the bike (that i do not have before) but nothing more.

i have always the message remote I/O error , write.
problem wih I2C and other settings of ttyUSB0?
I keep searching.

Laurent

**Comment by 360manu** [comment]
hello

I pushed a new version without i2c oled screen.
It used to display a message to log some info about the serial port.
But it was probably launching an exception while logging.

It doesn't mean that it works.
But you can try.



Le mer. 24 juin 2020 à 19:57, laurent-web <notifications@github.com> a
écrit :

**Comment by thesoulpole** [comment]
Hello,
I am trying to get your bridge to work with my Kettler ergometer FX1 (I am hoping that since you say it works with EX1, it will also work with my FX1). I bought an R Pi Zero WH.
Great of you to have made this bridge, by the way ! :slight_smile:
I have a question. I have gone through steps including the npm install, which I believe succeeded (lots of messages).
BUT - how do I connect the Kettler AND the keyboard at the same time? It has only one USB port.
Now, without bike connected and the keyboard connected, when I run “sudo node server.js” I get:
“[KettlerUSB] port is not open, retry in 10s” looping until I Ctrl-break
I assume that is because the bike is not connected?
But if I connect the bike, how do I type in “sudo node server.js” since I will not have the keyboard connected (due to lack of a second USB)?

2 - second item is in a separate thread/issue

Thanks for any help and explanation!

**Comment by 360manu** [comment]
Hello

I would suggest working headless on the raspberry pi zero.
it means using a remote connection with another PC.
start step 3 :
https://desertbot.io/blog/headless-pi-zero-w-wifi-setup-windows

Or you can use a USB Hub, but I never tested this way.

Then using a remote connection, you will be able to connect the USB Cable
directly to the Kettler.
And KettlerUSB will try to open the serial port

Le mer. 7 oct. 2020 à 18:55, Piotr Sienkiewicz <notifications@github.com> a
écrit :

**Comment by thesoulpole** [comment]
Thanks a lot.
In the meantime I managed to connect my Apple Bluetooth kbd (and mouse), over BT. So the USB is now free.
When my SUB-RS232 cable arrives I will see if this works.
If not, then headless (thanks for the link, it has a Mac version, luckily for me) is certainly a promising option.

Any chance on some help from you on my problem 2, which I wrote in the new Issue (Symlink not creating) I opened? 
Thanks in advance, much appreciated.

**Comment by thesoulpole** [comment]
Hello, 
Cable arrived, it seems to connect to the bike because bike screen displays a RS232 icon. 
However - it goes into a infinite loop - see attached photo
any ideas on what to do?
thanks in advance!

![IMG_3215](https://user-images.githubusercontent.com/28165922/95492834-f5010e00-099b-11eb-9779-1c7d165e4144.JPG)
![IMG_3217](https://user-images.githubusercontent.com/28165922/95492864-faf6ef00-099b-11eb-91c2-dc7fdfc333eb.JPG)

**Comment by 360manu** [comment]
Good ! it works !
It' displays a log status that it receives from the bike
the 25 on the line is your wattage

**Comment by 360manu** [comment]
Now you should try connect it to Zwift or any FTMS BLE compatbile software (I saw a game on Google Play)

**Comment by thesoulpole** [comment]
Hey, you know what? It WORKS! ;))
I connected no problem!

Now I have to learn this Zwift (and see what actually works, and if there are any missing bits and if they are of interest)

Will let you know!
